src/EnergyPlus/ExhaustAirSystemManager.cc

```diff
@@ -0,0 +1,966 @@
+                if (thisExhSys.SizingFlag) {
+                    SizeExhaustSystem(state, exhSysNum);
+                }
+            }
+            state.dataZoneEquip->NumExhaustAirSystems = numExhaustSystems;
+
+        } else {
+            // If no exhaust systems are defined, then do something <or nothing>:
+        }
+
+        if (ErrorsFound) {
+            ShowFatalError(state, "Errors found getting AirLoopHVAC:ExhaustSystem.  Preceding condition(s) causes termination.");
+        }
+    }
+
+    void InitExhaustAirSystem([[maybe_unused]] int &ExhaustAirSystemNum) // maybe unused
+    {
+    }
+
+    void CalcExhaustAirSystem(EnergyPlusData &state, int &ExhaustAirSystemNum, bool FirstHVACIteration)
+    {
+        auto &thisExhSys = state.dataZoneEquip->ExhaustAirSystem(ExhaustAirSystemNum);
+        // 2022-01: Simulate Zone Air Mixer
+        if (!(state.dataAirflowNetwork->AirflowNetworkFanActivated &&
+              state.dataAirflowNetwork->SimulateAirflowNetwork > AirflowNetwork::AirflowNetworkControlMultizone)) {
+            EnergyPlus::MixerComponent::SimAirMixer(state,
+                                                    state.dataZoneEquip->ExhaustAirSystem(ExhaustAirSystemNum).ZoneMixerName,
+                                                    state.dataZoneEquip->ExhaustAirSystem(ExhaustAirSystemNum).ZoneMixerIndex);
+        } else {
+            // 2022-02: Give a warning that the current model does not work with AirflowNetwork for now
+        }
+
+        Real64 mixerFlow_Prior = 0.0;
+        int outletNode_index = state.dataMixerComponent->MixerCond(thisExhSys.ZoneMixerIndex).OutletNode;
+        mixerFlow_Prior = state.dataLoopNodes->Node(outletNode_index).MassFlowRate;
+        if (mixerFlow_Prior == 0.0) {
+            // No flow coming out from the exhaust controls;
+            // fan should be cut off now;
+        }
+
+        // 2022-01-26: One additional step here might be to consider the avaiability schedule of the exhasut system
+        // or the central exhaust fan's own avaiability schedule.
+        // Need to prodeed differently for the cases when the exhasut system (fan) is available or not available.
+
+        // 2022-01: Simulate the fan (need some clean up 2022-01-26)
+        // calculate fan speed ratio for Fan:OnOff or Fan:SystemModel (not used for other fan types).
+        Real64 FanAirVolFlow = 1.0; // 2022-01: this should be something like a design or rate fan flow rate
+        Real64 FanSpeedRatio = 1.0; //  Node(InletNode).MassFlowRate / (FanAirVolFlow * state.dataEnvrn->StdRhoAir);
+
+        // Constant fan and variable flow calculation AND variable fan
+        auto &ZoneCompTurnFansOff = state.dataHVACGlobal->ZoneCompTurnFansOff;
+        auto &ZoneCompTurnFansOn = state.dataHVACGlobal->ZoneCompTurnFansOn;
+
+        int outletNode_Num = 0;
+
+        if (state.dataZoneEquip->ExhaustAirSystem(ExhaustAirSystemNum).CentralFanTypeNum == DataHVACGlobals::FanType_SystemModelObject) {
```

Comment:
should use the `thisExhSys` shortcut you created throughout here.

Reply:
This is done as well.